charmhub_lp_tools/reports.py

```diff
@@ -0,0 +1,207 @@
+import collections
+import json
+import logging
+import operator
+import os
+
+from datetime import datetime
+from typing import List
+try:
+    from zoneinfo import ZoneInfo
+except ImportError:
+    from backports.zoneinfo import ZoneInfo
+
+import humanize
+
+from jinja2 import Environment, FileSystemLoader, select_autoescape
+from prettytable import PrettyTable
+
+from .launchpadtools import TypeLPObject
+
+
+__THIS__ = os.path.dirname(os.path.abspath(__file__))
+NOW = datetime.now(tz=ZoneInfo("UTC"))
+
+
+class BaseBuildsReport:
```

Comment:
Might be worth deriving from [abc.ABC](https://docs.python.org/3/library/abc.html#abc.ABC)  - i.e. enforce that it's an abstract base class?

Reply:
ack, +1